examples/blocksub/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/flashbots/goutils/blocksub"
+)
+
+var (
+	httpURI  = os.Getenv("ETH_HTTP") // usually port 8545
+	wsURI    = os.Getenv("ETH_WS")   // usually port 8546
+	logJSON  = os.Getenv("LOG_JSON") == "1"
+	logDebug = os.Getenv("DEBUG") == "1"
+)
+
+func main() {
+	logLevel := log.LvlInfo
+	if logDebug {
+		logLevel = log.LvlDebug
+	}
+
+	logFormat := log.TerminalFormat(true)
+	if logJSON {
+		logFormat = log.JSONFormat()
+	}
+
+	log.Root().SetHandler(log.LvlFilterHandler(logLevel, log.StreamHandler(os.Stderr, logFormat)))
+	ch := make(chan *ethtypes.Header)
+	bs := blocksub.NewBlockSub(context.Background(), httpURI, wsURI, ch)
+	err := bs.Start()
+	if err != nil {
+		panic(err)
```

Comment:
panic(err) ?? would be good to use error or defer function
`defer func() {
		if err := recover(); err != nil {
			log.Println("panic occurred:", err)
		}
	}()`

Reply:
> panic(err) ?? would be good to use error or defer function `defer func() { if err := recover(); err != nil { log.Println("panic occurred:", err) } }()`

I'm not sure what you are suggesting.
I think it's good to `panic(err)` and exit since if this happens it's 99% something wrong with the configuration and the dev/devops need to know straight away. Panic's error formatting should be just fine, I'm not sure if you need to recover and log the error in a different format. Or are you suggesting something else?
i guess log.Crit could also work.

Reply:
this is the cli that should crash at this point, so panic is fine here. the program should not continue!

`recover` seems to be used only in http request context. what would this add here?

Reply:
my bad. I was thinking of recovering the panic..then I see that its start of the call.I will close the conversation.